fastanime/cli/interfaces/anilist_interfaces.py

```diff
@@ -573,15 +576,25 @@ def provide_anime(config: "Config", anilist_config: QueryDict):
             return
     anilist_config.anime_title = anime_normalizer.get(anime_title) or anime_title
     anilist_config._anime = search_results[anime_title]
-    fetch_anime_episode(config, anilist_config)
 
+    return config, anilist_config
 
 def anilist_options(config, anilist_config: QueryDict):
     selected_anime: "AnilistBaseMediaDataSchema" = anilist_config.selected_anime_anilist
     selected_anime_title: str = anilist_config.selected_anime_title
     progress = (selected_anime["mediaListEntry"] or {"progress": 0}).get("progress", 0)
     episodes_total = selected_anime["episodes"] or "Inf"
 
+    def _continue_watching(config: Config, anilist_config: QueryDict):
+        config.continue_from_history = True
+        config, anilist_config = provide_anime(config, anilist_config)
+        fetch_anime_episode(config, anilist_config)
+
```

Comment:
I will accept this changes and merge the later, though the tests should pass and you have made a slight error the provider fnction does not return any values

Reply:
I thought I changed provide_anime to return config and anilist_config.

Reply:
In any case did a whole lot of refactoring today since i felt like you and maybe other contributors will probably have a hard time reading my code especially anilist interface which is almost a 1000 lines of code. so i have added this personally as per your request, sorry i wasted your effort but i really felt the refactoring was important to do as soon as possible